lib/validator.js

```diff
@@ -167,20 +170,49 @@ Validator.prototype.compileSchemaType = function(schemaType) {
 	}
 
 	return this.compileSchemaRule(schemaType);
+
+};
+
+Validator.prototype.compileMessages = function(schemaType) {
+
+	let messages = {};
+
+	if (schemaType.messages && typeof schemaType.messages == "object")
+		messages = schemaType.messages;
+
+	if (!("required" in messages))
+		messages["required"] = this.messages["required"];
+
+	if (!(schemaType.type in messages))
+		messages[schemaType.type] = this.messages[schemaType.type];
+
+	Object.keys(schemaType).forEach(rule => {
+		if (rule == "type" || rule == "items" || rule == "props" || rule == "messages") return;
+
+		let key = schemaType.type.toLowerCase() + rule[0].toUpperCase() + rule.substring(1);
```

Comment:
My problem is that it is not guaranteed that you can generate message key from type & prop name (e.g `enumValue`)

Reply:
oooh.. almost.

Any suggestions?

Reply:
I don't understand exactly this parts, but can you forEach keys in the original `messages`? E.g. `Object.keys(this.messages).forEach(key => messages[key] = schemaType.messages[key] || this.messages[key]`?

Reply:
shouldn't that key just be 'enum'?

Reply:
and we add stringEnum etc

Reply:
it can be, just it caused a breaking change.

Reply:
that would work, it would mean passing all of the messages along with every rule. Not sure how significant that is and how you feel about that?

Reply:
Yes, but I don't think it can cause any problem. Just some more bytes in the memory.

Reply:
ok since my linting is bad do you want to make update?

Reply:
or do I need to? This is my first pull request!

Reply:
I added editorconfig to this PR and fix lintings. Pull these changes and check your editor use linting based on editorconfig. E.g. your commit convert tabs to spaces but the repo uses tabs in source files.